### DISCUSSION TITLE
Too many MBs downloaded (perhaps)

### DISCUSSION THREAD

**Discussion by thistlillo** [Homebrew/discussions#4179] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 3.6.20
ORIGIN: https://github.com/Homebrew/brew
HEAD: 344d32bf7f40183359592ef81ca6486e0587981b
Last commit: 3 weeks ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 07702893543bdce479e21e7ea2c9003d83c38754
Core tap last commit: 46 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I have a curiosity.

From this page: https://fishshell.com it is possible to download the fish shell for several operating systems, including Mac OS. For Mac, it is possible to download a self-contained app that is ~30MB in size.

However, I installed the fish shell via homebrew: it downloaded plenty of stuffy, including openjdk and billions of libraries. 

I actually do not know how many MBs it downloaded, but the only idk, that does not seem so necessary considering that the app is ~30MB, is more than 300MB.

Is there any way to become aware of the size of the packages, that:

1. a user is about to install (so BEFORE installation)

2. a user has installed (after INSTALLATION)

3. How are all the dependencies established? How can a 30MB standalone app result in hundreds of megabytes of brew casks?

**Comment by SMillerDev** [comment]
> How are all the dependencies established?

They are defined by upstream 

> How can a 30MB standalone app result in hundreds of megabytes of brew casks?

https://docs.brew.sh/FAQ#why-does-brew-upgrade-formula-or-brew-install-formula-also-upgrade-a-bunch-of-other-stuff

**Comment by thistlillo** [comment]
Thank you, I did read something about that. But I have the HD icon on the desktop showing the free space and it went down 1 GB. I think I can only ask to the maintainer of fish on homebrew.

**Reply by gromgit** [reply]
> But I have the HD icon on the desktop showing the free space and it went down 1 GB.

The link @SMillerDev posted explains why. Basically, installing or upgrading _any_ formula will also update all of its outdated dependencies, then update all outdated formulae that depend on those just-updated formulae, etc. so forth.

That's the price of using a package ecosystem like Homebrew that's both evergreen and rigorously release-tested.

**Reply by thistlillo** [reply]
May be that's the reason: may be new dependencies were added to packages that got upgraded. I have found this: https://github.com/Homebrew/discussions/discussions/57

that might explain what happened.

**Comment by carlocab** [comment]
Installing `fish` through `brew` requires ~40MB of space. `fish` itself is about 20MB, and depends on `ncurses` (~15MB) and `pcre2` (~7MB).
